File: src/main/java/de/huddeldaddel/euler/Problem3.java
package de.huddeldaddel.euler;

import de.huddeldaddel.euler.math.Factorizer;
import de.huddeldaddel.euler.math.Factorizer;

import java.util.ArrayList;
import java.util.List;

/**
 * Solution for https://projecteuler.net/problem=3
 */
public class Problem3 {

    public static void main(String[] args) {
        final List<Long> result = new ArrayList<>();
        new Factorizer().getPrimeFactors(600_851_475_143L, result);
        System.out.println(result.get(result.size() -1));
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem3.kt
package de.huddeldaddel.euler

import de.huddeldaddel.euler.math.Factorizer

/**
 * Solution for https://projecteuler.net/problem=3
 */
fun main() {
    val result = mutableListOf<Long>()
    Factorizer().getPrimeFactors(600_851_475_143L, result)
    println(result[result.size - 1])
}
